### PR TITLE
[Bugfix] Fix Wan spatial reshard boundary

### PR DESCRIPTION
**Pull request by rahul-steiger-nv**
## Purpose

Fix Wan spatial-shard VAE decoding when padding creates an empty trailing shard.

After gathering and trimming the global tensor, the last rank could compute a start offset beyond the trimmed extent. `torch.narrow` rejects this offset even when the requested length is zero.

Clamp the start and valid extent to the actual tensor size, then preserve equal shard shapes with zero-padding. Add diagnostics for rank and attention extent mismatches.

## Test Plan

- Run CPU spatial-shard unit tests.
- Compare height/width sharded decode against the reference on multiple GPUs.
- Run a 4-GPU Cosmos3 video smoke test.

**vLLM Version:** 0.25.0

**vLLM-Omni Commit:** 688143b199e4ebde1fce6e8d29cabb10d5c3e94a

## Test Result

- CPU: `25 passed, 2 deselected`
- Multi-GPU reference comparison: `2 passed`
- 4-GPU Cosmos3 requests:
  - 80x64 boundary regression: HTTP 200
  - 320x192, 189 frames: HTTP 200

### CODE REVIEW

File: vllm_omni/diffusion/distributed/autoencoders/wan_spatial_shard.py [hunk 1]
```diff
@@ -203,8 +203,17 @@ def reshard_from_trimmed_extent(
...
+    if ctx is not None and ctx.rank != rank:
+        raise RuntimeError(
+            "Wan VAE spatial-shard rank mismatch while resharding: "
+            f"group_rank={rank}, context_rank={ctx.rank}, split_dim={split_dim!r}."
+        )
+
     valid_extent = _local_valid_extent(local_extent)
-    start = rank * local_extent
+    actual_extent = x.shape[dim]
+    start = min(rank * local_extent, actual_extent)
```
Comment (hsliuustc0106) [inline vllm_omni/diffusion/distributed/autoencoders/wan_spatial_shard.py:215]: This handles the empty-tail case: when `rank * local_extent` exceeds the trimmed extent, clamping to `actual_extent` makes the zero-length `narrow` legal while the existing padding restores the fixed shard size.

File: tests/diffusion/distributed/test_wan_spatial_shard.py [hunk 2]
```diff
@@ -231,6 +231,128 @@ def test_reshard_from_trimmed_width_pads_invalid_columns(monkeypatch: pytest.Mon
     assert torch.equal(out.flatten(), torch.tensor([4.0, 0.0]))
 
 
+@pytest.mark.core_model
+@pytest.mark.cpu
+@pytest.mark.parametrize("split_dim", ["height", "width"])
+@pytest.mark.parametrize(
+    ("full_extent", "world_size"),
+    [
+        pytest.param(5, 4, id="four-ranks-empty-tail"),
```
Comment (hsliuustc0106) [inline tests/diffusion/distributed/test_wan_spatial_shard.py:240]: This 4-rank case reproduces the failure on #6013's current head; parametrizing both spatial axes also protects its dynamic height/width selection when that PR rebases.

**Review by hsliuustc0106 — APPROVED** [review]
LGTM. Verified the focused CPU suite and 2-GPU height/width parity tests.
